meson.build

```diff
@@ -0,0 +1,103 @@
+
+gnome = import('gnome')
+
+# Build options
+
+jasmine_report_argument = ''
+junit_reports_dir = get_option('jasmine_junit_reports_dir')
+if junit_reports_dir != ''
+    jasmine_report_argument = '--junit ' + junit_reports_dir
+endif
+
+# Paths
+
+abs_libexecdir = join_paths(get_option('prefix'), get_option('libexecdir'))
+
+# Dependencies
+
+webkit = dependency('webkit2gtk-4.0')
+javascriptcore = dependency('javascriptcoregtk-4.0')
+webkit_webextension = dependency('webkit2gtk-web-extension-4.0')
+
+# Javascript module
+
+javascript_sources = [
+    'webhelper/webhelper.js',
+    'webhelper/webhelper2.js',
+]
+gjs_modules_dir = join_paths(get_option('datadir'), 'gjs-1.0')
+install_data(javascript_sources, install_dir: gjs_modules_dir)
+
+js_config = configuration_data()
+js_config.set('libexecdir', abs_libexecdir)
+configure_file(input: 'webhelper/webhelper_private/config.js.in',
+    output: 'config.js', configuration: js_config, install: true,
+    install_dir: join_paths(gjs_modules_dir, 'webhelper_private'))
+
+# Private C library
+
+# Workaround for https://bugs.webkit.org/show_bug.cgi?id=116672
+# When that is solved, we can eliminate this private library and go back to
+# using pure Javascript in WebHelper.
+private_sources = [
+    'webhelper/lib/wh2private.c',
+    'webhelper/lib/wh2private.h',
+]
+private_lib = library('webhelper2private', private_sources,
+    dependencies: webkit)
+gnome.generate_gir(private_lib, sources: private_sources, nsversion: '1.0',
+    install: true, namespace: 'WebHelper2Private', symbol_prefix: 'wh2_',
+    identifier_prefix: 'Wh2', includes: ['WebKit2-4.0'],
+    extra_args: ['--pkg=webkit2gtk-4.0'])
+
+# Web extension
+
+web_extension_sources = [
+    'webhelper/webextensions/wh2extension.c',
+    'webhelper/webextensions/wh2jscutil.c',
+    'webhelper/webextensions/wh2jscutil.h',
+]
+web_extension_dir = join_paths(get_option('libexecdir'), 'webhelper2')
+shared_module('wh2extension', web_extension_sources, name_prefix: '',
+    dependencies: [javascriptcore, webkit_webextension], install: true,
+    install_dir: web_extension_dir)
+
+# Tests
+
+jasmine = find_program('jasmine')
+test_runner = find_program('./test/tap.py')
+javascript_tests = [
+    'test/webhelper/testLocal.js',
+    'test/webhelper/testTranslate.js',
+    'test/webhelper/testTranslate2.js',
+    'test/webhelper/testWebActions.js',
+    'test/webhelper/testWebActions2.js',
+    'test/webhelper/testUpdateFontSize.js',
+]
+include_path = join_paths(meson.current_source_dir(), 'webhelper')
+tests_environment = environment()
+tests_environment.prepend('GJS_PATH', include_path)
+tests_environment.prepend('GI_TYPELIB_PATH', meson.current_build_dir())
+tests_environment.prepend('LD_LIBRARY_PATH', meson.current_build_dir())
+tests_environment.set('WEBHELPER_UNINSTALLED_EXTENSIONS_DIR',
+    meson.current_build_dir())
+tests_environment.set('TOP_SRCDIR', meson.current_source_dir())
+
+args = [jasmine.path()[0], '--no-config', '--tap']
```

Comment:
It's likely; I've opened https://github.com/mesonbuild/meson/issues/1547 to track it.